docs/content/sips/xxx-spin-plugins.md

```diff
@@ -0,0 +1,154 @@
+title = "SIP xxx - Plugin System for Spin"
+template = "main"
+date = "2022-08-09T13:22:30Z"
+---
+
+Summary: Plugin system for spin.
+
+Owner: karthik.ganeshram@fermyon.com
+
+Created: August 9, 2022 
+
+## Background 
+
+As the functionality fo spin gets extended, there will be a point where every feature will not be required/used by every user. An example of this would be different triggers for spin along with possible subcommands to add more functionality to spin. Therefore it would make sense to be able to add features as plugins allowing for addition of only the required features witout modifying the spin binary.
+
+## Proposal
+
+Create a new subcommand for spin called `spin plugin` which will have two further subcommands namely
+
+- `install`
+- `uninstall`
+
+### Types of plugins
+
+![Classification of types of plugins](https://i.imgur.com/Fo7EGPQ.png)
+
+- Standalone Binary Plugins (external subcommands) - These work to add features to spin and do not need any use of the spin binary.
```

Comment:
if both Internal Spin Plugins and Spin Dependent User Plugins fall under the "Spin Dependent" category, then maybe they should both be prefixed by "Spin Dependent". IE Internal Spin Plugins => Spin Dependent Internal Plugin

Reply:
Will make the changes.